### PR TITLE
OS/2 port of FeatherPad

### PR DESCRIPTION
**Pull request by josch1710**
To have a lightweight editor for porting stuff, I also ported FeatherPad to OS/2.

### CODE REVIEW

**Comment by tsujan** [comment]
Thanks!

I can't test it under OS/2. Does everything work fine? Hard-coded icons, saving of settings,... Spell checker too?

File: featherpad/fpwin.cpp [hunk 1]
```diff
@@ -695,7 +695,7 @@ void FPwin::applyConfigOnStarting()
         config.setReservedShortcuts (reserved);
         config.readShortcuts();
     }
-
+
```
Comment (tsujan) [inline featherpad/fpwin.cpp:698]: Would you please remove this redundant change and also put the OS/2-specific file `featherpad.rc` inside the `data` directory?

**Comment by josch1710** [comment]
Hi.
I removed the tab and moved featherpad.rc to data. I also deactivated the -mtune flags for now, because something doesn't work out outside rpmbuild. I have to investigate it.

As for the port itself: Icons are there, settings are saved etc. The only thing, which doesn't activate at the moment, is the Run menu item. I still have to investigate that. 
[Screenshot](https://www.os2world.com/forum/index.php?action=dlattach;topic=2859.0;attach=7679;image)

**Comment by tsujan** [comment]
Thanks for the changes!

I also had a question about the SVG icon. It's used with qmake but not with cmake. Is that intentional?

**Comment by josch1710** [comment]
No, I just finished the port using cmake, because that works better.
I have to look at the qmake stuff again for more parity.

**Comment by tsujan** [comment]
> The only thing, which doesn't activate at the moment, is the Run menu item. I still have to investigate that.

Maybe, on OS/2, executable files cannot be executed only by entering their paths. FeatherPad has a setting for the command in Preferences → Files.

**Comment by tsujan** [comment]
> I have to look at the qmake stuff again for more parity.

OK. Please tell me when your work is finished. I want to merge this.

**Comment by josch1710** [comment]
> > The only thing, which doesn't activate at the moment, is the Run menu item. I still have to investigate that.
> 
> Maybe, on OS/2, executable files cannot be executed only by entering their paths. FeatherPad has a setting for the command in Preferences → Files.

Thanks for the hint. I haven't explored all nooks and crannies of FeatherPad, yet ;-)
But otherwise, the port was a breeze.

**Comment by josch1710** [comment]
So, "Run" works with the setting. Thanks.

I reworked the qmake project and renamed featherpad.rc to featherpad_os2.rc to be clear, that the file is used for OS/2 and not Windows. Also it's more with featherpad_os2.ico.

I found two typos in the german translation file and fixed them.
I'm finished for now ;-)

**Comment by tsujan** [comment]
Thank you very much!

I'll add a line about OS/2 to `README.md` later. Merging…

**Comment by josch1710** [comment]
I also have binaries in my release section.

**Comment by tsujan** [comment]
> I also have binaries in my release section.

If FeatherPad's `INSTALL` file was as detailed as [that of Kvantum](https://github.com/tsujan/Kvantum/blob/master/Kvantum/INSTALL.md), I would add a link to it but `INSTALL` is simple and generic.

**Comment by josch1710** [comment]
It's fine. Just wanted to let you know.

**Comment by tsujan** [comment]
@josch1710
Could you please check that the qmake compilation works with the latest git FeatherPad? — after this commit: https://github.com/tsujan/FeatherPad/commit/0132aeb95ce727feb3f3feb97ccb9355f84b46c0

**Comment by josch1710** [comment]
@tsujan 
I have made a pull request.

**Pull request merged**